bundle/config/mutator/resolve_variable_references.go

```diff
@@ -181,13 +181,12 @@ func (m *resolveVariableReferences) Apply(ctx context.Context, b *bundle.Bundle)
 		// Normalize the result because variable resolution may have been applied to non-string fields.
 		// For example, a variable reference may have been resolved to a integer.
 		root, diags := convert.Normalize(b.Config, root)
-		for _, diag := range diags {
-			// This occurs when a variable's resolved value is incompatible with the field's type.
-			// Log a warning until we have a better way to surface these diagnostics to the user.
-			log.Warnf(ctx, "normalization diagnostic: %s", diag.Summary)
-		}
+		allDiags = allDiags.Extend(diags)
 		return root, nil
 	})
 
-	return diag.FromErr(err)
+	if err != nil {
+		allDiags = allDiags.Extend(diag.FromErr(err))
+	}
+	return allDiags
```

Comment:
Could you flip this around and name it `diags` here at the top level and use a different name inner?

Then the naming pattern will be the same as elsewhere.